### PR TITLE
Expanded build matrix

### PR DESCRIPTION
**Pull request by aaronpowell**
This is for issue #187, showing how it could be possible to expand the build matrix.

A point of note, this contains the changes from PR #169 as well, since that hit the issue of #187, so the intent is to show how it can be addressed.

### CODE REVIEW

**Comment by davidmrdavid** [comment]
This looks great! Thanks for doing this, it saved us the effort :) . So, from what I gather, the trick was pretty much to disable linting during the Node8 tests, which I guess would have depended on our need `Prettier` dependency, huh?

In any case, this PR makes sense to me. Since it changes a lot of things though, I'll just wait for another thumbs up from either @ConnorMcMahon or @cgillum before giving it the approval and merging.

**Comment by aaronpowell** [comment]
Pretty much @davidmrdavid, I've expanded the matrix and if it's Node.js v8 I created a few new `npm run-script` options that just drop linting and the others all do it still.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
-  - task: NodeTool@0
-    inputs:
-      versionSpec: $(NODE_VERSION)
-    displayName: 'Install Node.js'
-  - script: npm install
-    displayName: 'npm install'
-  - script: npm run test
-    displayName: 'npm build and test'
+variables: {
+  MODULE_VERSION: '1.2.2',
```
[the remark this answers is not in the document]
Comment (aaronpowell) [reply azure-pipelines.yml:2]: Easiest way to do it is to run `node -e 'console.log(require("./package.json").version)'` and have that in a script that you create a pipeline variable from.

[the remark this answers is not in the document]
Comment (davidmrdavid) [reply azure-pipelines.yml:2]: Perfect, I'll make this into an issue so we take care of it before the next release, I'll merge this now

**Comment by davidmrdavid** [comment]
This was an **awesome** PR, @aaronpowell . Many thanks! ⚡ ⚡

File: azure-pipelines.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidmrdavid) [reply azure-pipelines.yml:2]: Now tracked in https://github.com/Azure/azure-functions-durable-js/issues/194

**Pull request merged**